src/components/ImageListContainer.js

```diff
@@ -0,0 +1,79 @@
+import React from "react";
+import {addImages, endDowloadingImages, startDowloadingImages} from "../actions";
+import {connect} from "react-redux";
+import ImageList from "./ImageList";
+import PropTypes from "prop-types";
+import {IMAGE_HEIGHT} from "../constants";
+
+class ImageListContainer extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            isFetching: false,
+        };
+    }
+
+    handleScroll = (e) => {
+        const bottom = e.target.scrollingElement.scrollHeight - IMAGE_HEIGHT - e.target.scrollingElement.scrollTop <= e.target.scrollingElement.clientHeight + 1;
+        if (bottom&&!this.props.isFetching) {
+            this.props.addImagesToStore();
+        }
+    }
+
+    componentDidMount() {
+        this.props.addImagesToStore();
+        window.addEventListener("scroll", this.handleScroll);
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener("scroll", this.handleScroll);
+    }
+
+
+    render() {
+        return (
+            <ImageList data={this.props.images}/>
+        );
+    }
+
+}
+
+function mapStateToProps(state) {
+    return {
+        images: state.images,
+        isFetching: state.isFetching,
+    }
+}
+
+function mapDispatchToProps(dispatch) {
+    return {
+        addImagesToStore: () => dispatch(addCat()),
+    }
+}
+
+function addCat(){
+    return (dispatch) => {
+        dispatch(startDowloadingImages())
+        fetch("https://api.thecatapi.com/v1/images/search?limit=10")
+            .then((response) => {
+                return response.json();
+            })
+            .then((arrayOfData) => {
+                dispatch(addImages(arrayOfData));
+                dispatch(endDowloadingImages())
```

Comment:
From my experience, it may be really hard to remember considering the `catch` case whenever using a promise. This code is a good example for this issue. For example, If we don't get the HTTP response from the Cats API call (e.g. it's a network issue), then the 2 `then` functions above will _not_ be invoked, and as a result the `state.isFetching` field will be forever `true`. So, let's add a corresponding `catch` call.

Reply:
Also, for learning sake, let's add the handling of HTTP responses with error statuses. Whenever the Cats API call returns an HTTP response, `then` functions of the promise are invoked. You may use the `ok` field of the passed `response` object ([docs](https://developer.mozilla.org/en-US/docs/Web/API/Response/ok)) in order to distinguish between successful responses (2xx status) and unsuccessful ones (all other statuses).

Reply:
Resolved.